examples/ring_topology.rs

```diff
@@ -0,0 +1,42 @@
+use rr_channel::detthread;
+use rr_channel::crossbeam;
+use std::sync::Mutex;
+use std::sync::Arc;
+
+fn main() {
+    let mut senders: Vec<Mutex<_>> = Vec::new();
+    let mut receivers: Vec<Mutex<_>> = Vec::new();
+
+    for _ in 0..10 {
+        let (s, r) = crossbeam::unbounded::<i32>();
+        senders.push(Mutex::new(s));
+        receivers.push(Mutex::new(r));
+    }
+    
+    let senders = Arc::new(senders);
+    let receivers = Arc::new(receivers);
+
+    for i in 0..10 {
+        // Clone via arc for moving into closure.
+        let senders = senders.clone();
+        let receivers = receivers.clone();
+        rr_channel::detthread::spawn(move || {
+            //Set up the ring
+            if i == 0 {
+                let sender = senders[0].try_lock().expect("Lock already taken");
+                sender.send(1);
+            }
+
+            // Locks should never block as every thread accesses exclusive members of vector.
+            let sender = senders[(i + 1) % 10].try_lock().expect("Lock already taken");
+            let receiver = receivers[(i) % 10].try_lock().expect("Lock already taken");
```

Comment:
I would move these lines above the `if` to avoid having to `try_lock()` the sender twice.

Reply:
I think the sender in if statement is sender[0] and this sender starts from sender 1?

Reply:
Oh I see. Yes, then maybe just move lines 25-28 to the main thread before we create the threads then? It is a little confusing how we currently have it.